framework_tvb/tvb/adapters/creators/connectivity_creator.py

```diff
@@ -114,27 +149,27 @@ def _store_related_region_mappings(self, original_conn_gid, new_connectivity_ht)
 
         return result
 
-    def launch(self, original_connectivity, new_weights, new_tracts, interest_area_indexes, is_branch=False, **kwargs):
+    def launch(self, view_model):
         """
         Method to be called when user submits changes on the
         Connectivity matrix in the Visualizer.
         """
         # note: is_branch is missing instead of false because browsers only send checked boxes in forms.
-        original_conn_ht = h5.load_from_index(original_connectivity)
+        original_conn_ht = h5.load_from_index(view_model.original_connectivity)
         assert isinstance(original_conn_ht, Connectivity)
 
-        if not is_branch:
-            new_conn_ht = self._cut_connectivity(original_conn_ht, numpy.array(new_weights),
-                                                 numpy.array(interest_area_indexes), numpy.array(new_tracts))
+        if not view_model.is_branch:
+            new_conn_ht = self._cut_connectivity(original_conn_ht, numpy.array(view_model.new_weights),
+                                                 numpy.array(view_model.interest_area_indexes), numpy.array(view_model.new_tracts))
```

Comment:
all these numpy.array() should not be necessary as view_model attributes are already defined as NArray